src/storage/BaseRecord.ts

```diff
@@ -1,8 +1,17 @@
 import { Exclude, Type } from 'class-transformer'
 
-export type Tags = Record<string, string | undefined>
+export type TagValue = string | boolean | undefined
+export type TagsBase = {
+  [key: string]: TagValue
+  [key: number]: never
+}
+
+export type Tags<DefaultTags, CustomTags extends TagsBase> = CustomTags & DefaultTags
+
+export abstract class BaseRecord<DefaultTags = Record<string, unknown>, CustomTags extends TagsBase = TagsBase> {
```

Comment:
I’m sorry, I don’t want to bother with nitpicking, but I try to understand the typings here. Bear with me, please :)

Does it mean that `DefaultTags` allowing the data structures that are prohibited by `TagsBase`? 

Is the goal that `CustomTags` cannot have number as a key or value but `DefaultTags` can?

I would assume that `DefaultTags` should have the same type as `TagsBase`. Then, maybe, we could simplified it to this:

```typescript
export type TagValue = string | boolean | undefined
export type Tags = {
  [key: string]: TagValue
  [key: number]: never
}

export type RecordTags<DefaultTags, Tags> = Tags & DefaultTags

export abstract class BaseRecord<DefaultTags = Tags, CustomTags = Tags> {
```

Reply:
Of course, we can merge this if you want, I know it's been here for a while. I'm just curious if there is a way to simplify it a bit because it took me some time to understand what exactly we want to achieve with those types.

Reply:
I don't mind the nitpicking :)

You're totally right. I was struggling a bit with the types and wanted to indicate that by default he `DefaultTags` is an empty object. But with `Record<string, unknown>` that isn't any more clear. I changed it so both extend the `TagsBase` type now.